### PR TITLE
Replace old C headers (xxx.h) with modern ones (cxxx)

### PR DESCRIPTION
**Pull request by pi1024e**
## Summary of the Pull Request
Most files already have switched to use the modern variation, such as "", "", etc. However, a few files do not, so this PR fixes that.
 
## PR Checklist
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Tests added/passed

## Validation Steps Performed
- Unit Tests
- Followed instructions by the ISO Draft 2005 on C++

### CODE REVIEW

**Comment by ZhaoMJ** [comment]
`Abs` and `<cstdlib>` is somewhat problematic and I think that's why the tests failed. See also https://developers.redhat.com/blog/2016/02/29/why-cstdlib-is-more-complicated-than-you-might-think/.

**Comment by pi1024e** [comment]
Because of the extern C. Yeah turns out they aren't exactly interchangable because of the namespace and extern

**Comment by pi1024e** [comment]
Working on it.

File: dep/chromium/base/numerics/checked_math.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #ifndef BASE_NUMERICS_CHECKED_MATH_H_
 #define BASE_NUMERICS_CHECKED_MATH_H_
 
-#include <stddef.h>
+#include <cstddef>
```
Comment (skyline75489) [inline dep/chromium/base/numerics/checked_math.h]: I don't think we want to modify this. It's third party code.

File: src/interactivity/onecore/precomp.h [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include <ntrtl.h>
 #include <nturtl.h>
 #include <windows.h>
-#include "wchar.h"
+#include "cwchar"
```
Comment (skyline75489) [inline src/interactivity/onecore/precomp.h]: These are also code from outside the repo, I suppose.

**Comment by DHowett-MSFT** [comment]
Please don’t change code in the `dep` folder. Files in that folder are dependencies from other projects.

File: src/interactivity/onecore/precomp.h (hunk 2 above)
Comment (pi1024e) [reply src/interactivity/onecore/precomp.h]: Is that so?

**Comment by pi1024e** [comment]
Done @DHowett-M @zadjii-msft

**Comment by pi1024e** [comment]
So stdlib.h and corecrt_math conflict each other...

**Comment by pi1024e** [comment]
Fixed it @DHowett-MSFT @zadjii-msft, I had to pull the c-headers from the extern C block

**Comment by zadjii-msft** [comment]
@msftbot make sure @DHowett-MSFT and @miniksa sign off on this

**Comment by pi1024e** [comment]
@miniksa are these changes good to go?

**Comment by pi1024e** [comment]
@DHowett-MSFT

**Comment by miniksa** [comment]
@pi1024e, I appreciate your enthusiasm, but this isn't high on my priority list right now. We have crashes and bugs and errors that are required to ship 1.0 and this is code health rearrangement. Don't get me wrong, I love code health improvements. But I think this will probably land after 1.0 branch forks as it doesn't really *do* anything for the product.

**Comment by pi1024e** [comment]
Alright. I was just hoping this would land soon, especially since it was improved, but if you don't want to right now that is fine.

**Comment by DHowett-MSFT** [comment]
Thanks for understanding! We're really excited to have somebody helping us clean up our mess :smile:

**Comment by pi1024e** [comment]
@zadjii-msft Anything you want to say about this?

**Comment by pi1024e** [comment]
@zadjii-msft Anything else you would like to say about this?

**Pull request merged**